include/mbedtls/ssl.h

```diff
@@ -1114,31 +1091,31 @@ struct mbedtls_ssl_session
     unsigned int key_exchange; /* Indication of the key exchange algorithm being negotiated*/
     unsigned char key_exchange_modes; /*!< psk key exchange modes */
 #endif /* MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL */
-#if defined(MBEDTLS_SSL_SESSION_TICKETS) && defined(MBEDTLS_SSL_CLI_C)
-    unsigned char *ticket;      /*!< RFC 5077 session ticket */
+#if defined(MBEDTLS_SSL_SESSION_TICKETS) || defined(MBEDTLS_SSL_NEW_SESSION_TICKET)
+    unsigned char *ticket;      /*!< Ticket */
     size_t ticket_len;          /*!< session ticket length   */
     uint32_t ticket_lifetime;   /*!< ticket lifetime hint    */
-#endif /* MBEDTLS_SSL_SESSION_TICKETS && MBEDTLS_SSL_CLI_C */
+#endif /* MBEDTLS_SSL_SESSION_TICKETS && MBEDTLS_SSL_NEW_SESSION_TICKET */
 
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL) && defined(MBEDTLS_SSL_NEW_SESSION_TICKET)
-// TBD: Replace fields by ticket structure
-// We currently only store a single ticket on the client size
-    unsigned char* ticket;          /*!< TLS 1.3 session ticket acting as psk identity */
-    size_t ticket_len;              /*!< ticket length   */
-    uint32_t ticket_lifetime;       /*!< ticket lifetime hint    */
-    mbedtls_ssl_ticket_flags flags; /*!< ticket flags */
-    uint32_t ticket_age_add;        /* A randomly generated 32-bit value that is used to obscure the age of the ticket */
-    unsigned char* ticket_nonce;    /*!< ticket nonce value */
-    uint8_t ticket_nonce_len;       /*!< ticket nonce length */
-    size_t key_len;                 /*!< psk key length */
+    int minor_ver;                  /*!< Version information for selecting ticket format */
+    unsigned int endpoint : 1;      /*!< 0: client, 1: server               */
```

Comment:
Can we just use `ssl->conf->endpoint`? Is there any case that we may store ticket from both client and server?